include/eve/module/core/regular/impl/simd/x86/convert.hpp

```diff
@@ -10,8 +10,754 @@
 #include <eve/as.hpp>
 #include <eve/concept/value.hpp>
 #include <eve/detail/implementation.hpp>
-
 #include <type_traits>
 
-#include <eve/module/core/regular/impl/simd/x86/convert_128.hpp>
-#include <eve/module/core/regular/impl/simd/x86/convert_256.hpp>
+namespace eve::detail
+{
+  //================================================================================================
+  // convert: logical -> logical
+  //================================================================================================
+  template<real_scalar_value T, typename N, real_scalar_value U>
+  EVE_FORCEINLINE logical<wide<U, N>> convert_impl( EVE_SUPPORTS(sse2_)
+                                                  , logical<wide<T,N>> v, as<logical<U>> const &tgt
+                                                  ) noexcept
+  {
+    constexpr auto c_in   = categorize<wide<T, N>>();
+    constexpr auto c_out  = categorize<wide<U, N>>();
+
+    //==============================================================================================
+    // Trivial cases : same size or unsigned input
+    //==============================================================================================
+        if constexpr( sizeof(U) == sizeof(T) )
+    {
+      return bit_cast(v, as<logical<wide<U,N>>>{} );
+    }
+    else if constexpr( std::is_unsigned_v<T> )
+    {
+      return convert( bit_cast(v, as< logical<wide<std::make_signed_t<T>,N>> >{}),tgt);
+    }
+    //==============================================================================================
+    // 8 bits cases - Always slice
+    //==============================================================================================
+    else  if constexpr( sizeof(T) == 1 && sizeof(U) > 2 && N::value > 1)
+    {
+      return convert( convert(v,as<logical<std::int16_t>>{}), tgt);
+    }
+    //==============================================================================================
+    // 16 bits cases
+    //==============================================================================================
+    else  if constexpr( c_in == category::int16x8 && sizeof(U) == 1 )
+    {
+      return  _mm_packs_epi16(v, v);
+    }
+    else  if constexpr( c_in == category::int16x16 && sizeof(U) == 1 )
+    {
+      return _mm_packs_epi16(v.slice(lower_), v.slice(upper_));
+    }
+    //==============================================================================================
+    // 32 bits cases
+    //==============================================================================================
+    else  if constexpr( c_in == category::int32x4 && sizeof(U) < 4 )
```

Comment:
why not just a ladder in the beginning and then you don't have to worry about it.

Seems like that ladder is common across all platforms btw.